pallets/identity/src/lib.rs

```diff
@@ -1174,72 +1159,57 @@ impl<T: Trait> Module<T> {
 
         let auth = <Authorizations<T>>::get(&signer, auth_id);
 
-        let identity_data_to_join = match auth.authorization_data {
-            AuthorizationData::JoinIdentity(data) => Ok(data),
+        let permissions = match auth.authorization_data {
+            AuthorizationData::JoinIdentity(permissions) => Ok(permissions),
             _ => Err(AuthorizationError::Invalid),
         }?;
 
+        //Not really needed unless we allow identities to be deleted
         ensure!(
-            <DidRecords<T>>::contains_key(&identity_data_to_join.target_did),
+            <DidRecords<T>>::contains_key(&auth.authorized_by),
             "Identity does not exist"
         );
 
-        let master = Self::did_records(&identity_data_to_join.target_did).master_key;
+        Self::consume_auth(auth.authorized_by.clone(), signer.clone(), auth_id)?;
 
-        Self::consume_auth(Signatory::Account(master), signer.clone(), auth_id)?;
-
-        Self::unsafe_join_identity(identity_data_to_join, signer)
+        Self::unsafe_join_identity(auth.authorized_by, permissions, signer)
     }
 
     /// Joins an identity as signer
     pub fn unsafe_join_identity(
-        identity_data_to_join: JoinIdentityData,
+        target_did: IdentityId,
+        permissions: Vec<Permission>,
         signer: Signatory<T::AccountId>,
     ) -> DispatchResult {
+        T::ProtocolFee::charge_fee(
+            &Signatory::Identity(target_did),
+            ProtocolOp::IdentityAddSigningItemsWithAuthorization,
+        )?;
         if let Signatory::Account(key) = &signer {
-            if !Self::can_key_be_linked_to_did(key, SignatoryType::External) {
-                ensure!(
-                    Self::get_identity(key) == Some(identity_data_to_join.target_did),
-                    Error::<T>::AlreadyLinked
-                )
-            } else {
```

Comment:
I can't remember why but there might be a reason why we added this condition. It basically allows an existing key to accept the authorization without errors.